src/extensions/default/CodeFolding/main.js

```diff
@@ -316,6 +320,38 @@ define(function (require, exports, module) {
         if (editor._codeMirror.getOption("gutters").indexOf(GUTTER_NAME) === -1) {
             createGutter(editor);
             restoreLineFolds(editor);
+            // Watch mutations on code mirror gutters and ensure line numbers are added before fold gutter.
+            var config = {childList: true};
+            var gutters,
+                lineNumberIndex,
+                foldGutterIndex,
+                cm = editor._codeMirror,
+                guttersContainer = $(".CodeMirror-gutters", editor.getRootElement()),
```

Comment:
`editor.getRootElement()` is used here, on L342, L344 and L346 so it could / should be saved to an variable